life_expectancy/transformation_interface.py

```diff
@@ -0,0 +1,115 @@
+from abc import ABC, abstractmethod
+import pandas as pd
+from life_expectancy.loading_strategies import Region
+
+
+class Transformation(ABC):
+    @abstractmethod
+    def transform(self, data: pd.DataFrame) -> pd.DataFrame:
+        """Transform the input data and return the transformed data."""
+
+    def is_necessary(self, data: pd.DataFrame) -> bool:
+        """Test if transformation is necessry"""
+        return True
+
+
+class TransformationPipeline:
+    def __init__(self):
+        self.transformations = []
+
+    def add_transformation(self, transformation: Transformation):
+        self.transformations.append(transformation)
+
+    def transform(self, data: pd.DataFrame) -> pd.DataFrame:
+        for transformation in self.transformations:
+            if transformation.is_necessary(data):
```

Comment:
shouldn't you have a warning in the case the transformation is 'not necessary'?

Reply:
Added